infrastructure/src/main/java/org/corfudb/infrastructure/redundancy/PrefixTrimRedundancyCalculator.java

```diff
@@ -0,0 +1,96 @@
+package org.corfudb.infrastructure.redundancy;
+
+import com.google.common.collect.ImmutableList;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegment;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.NOT_TRANSFERRED;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.RESTORED;
+import static org.corfudb.runtime.view.Address.NON_ADDRESS;
+
+/**
+ * A redundancy calculator that reads the global trim mark before creating a list of
+ * segments to transfer.
+ */
+public class PrefixTrimRedundancyCalculator extends RedundancyCalculator {
+
+    private final CorfuRuntime runtime;
+
+    public PrefixTrimRedundancyCalculator(String node, CorfuRuntime runtime) {
+        super(node);
+        this.runtime = runtime;
+    }
+
+    /**
+     * Sets the trim mark on this log unit and also performs a prefix trim.
+     *
+     * @param layout   A current layout.
+     * @param runtime  A current runtime.
+     * @param endpoint A current endpoint.
+     * @return A retrieved trim mark.
+     */
+    long setTrimOnNewLogUnit(Layout layout, CorfuRuntime runtime,
+                             String endpoint) {
+
+        long trimMark = runtime.getAddressSpaceView().getTrimMark().getSequence();
+
+        Token prefixToken = new Token(layout.getEpoch(), trimMark - 1);
+        runtime.getLayoutView().getRuntimeLayout(layout)
+                .getLogUnitClient(endpoint)
+                .prefixTrim(prefixToken)
+                .join();
+        return trimMark;
+    }
+
+    /**
+     * Given a layout, creates an initial list of non-empty and bounded transfer segments.
+     *
+     * @param layout A current layout.
+     * @return A list of transfer segments.
+     */
+    public ImmutableList<TransferSegment> createStateList(Layout layout) {
+        long trimMark = setTrimOnNewLogUnit(layout, runtime, getServer());
+        return layout.getSegments()
+                .stream()
+                // filter the segments that end before the trim mark and are not open.
```

Comment:
nit -> before -> after

Reply:
Maybe it's easier to say.. Keep all segments after the trim mark, except the open segment. (last segment).